nvflare/tool/code_pre_installer/install.py

```diff
@@ -0,0 +1,183 @@
+
+    Args:
+        application_dir: Base application directory (containing app directories)
+        site_name: Target site name to find app directories for
+
+    Returns:
+        Dictionary mapping job names to their app directory paths
+
+    Raises:
+        ValueError: If no matching app directories found for site
+    """
+    # Find all app directories that have meta.json
+    job_dirs = [d for d in application_dir.iterdir() if d.is_dir() and (d / "meta.json").exists()]
+    if not job_dirs:
+        raise ValueError("No application directories with meta.json found")
+
+    # Dictionary to store job_name -> app_dir mappings
+    matched_apps = {}
+
+    # Search through all app directories for matching deployments
+    for job_dir in job_dirs:
+        try:
+            with open(job_dir / "meta.json") as f:
+                meta = json.load(f)
+        except (FileNotFoundError, json.JSONDecodeError):
+            continue  # Skip invalid meta.json
+
+        deploy_map = meta.get("deploy_map", {})
+        if not deploy_map:
+            continue  # Skip if no deploy map
+
+        # Case 1: Site-specific format (app_site-1 -> [site-1, ...])
+        for app_name, sites in deploy_map.items():
+            if site_name in sites:
+                site_app_dir = job_dir / app_name
+                if site_app_dir.exists():
+                    matched_apps[job_dir.name] = site_app_dir
+
+        # Case 2: Default format (any_app_name -> [@ALL])
+        for app_name, sites in deploy_map.items():
+            if "@ALL" in sites:
+                default_app_dir = job_dir / app_name
+                if default_app_dir.exists():
+                    matched_apps[job_dir.name] = default_app_dir
+
+    if not matched_apps:
+        raise ValueError(f"No application directories found for site {site_name}")
+
+    return matched_apps
+
+
+def install_app_code(app_code_zip: Path, install_prefix: Path, site_name: str, target_shared_dir: str) -> None:
+    """Install application code from zip file.
+
+    Args:
+        app_code_zip: Path to application code zip file
+        install_prefix: Installation prefix directory
+        site_name: Target site name
+        target_shared_dir: Target shared directory path
+    """
+    if not app_code_zip.exists():
+        raise FileNotFoundError(f"Application code zip not found: {app_code_zip}")
+
+    # Create temp directory for extraction
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_path = Path(temp_dir)
+
+        # Extract zip
+        with ZipFile(app_code_zip) as zf:
+            zf.extractall(temp_path)
+
+        # Verify structure
+        application_dir = temp_path / "application"
+        shared_dir = temp_path / "application-share"
+        if not application_dir.exists() or not any(application_dir.iterdir()):
+            raise ValueError("Invalid application code: Missing application or application-share directory")
```

Comment:
is "application-share" a must have?
if it is then modify the "if" condition to check for that.
if it is NOT, then modify the error message.